### PR TITLE
Prepublish Checklist: Add check for video posters

### LINKED ISSUE #7613: Add check for missing video posters to the pre-publish checklist
## Bug Description

As part of the publishing checklist, we should ensure that every video in the story has an associated poster image. This is a required field. 

If a poster doesn't have a video poster, we should make it easy for the creator to re-trigger the video poster generation logic that we have in the editor today. 

Copy for the pre-publish checklist [is here](https://docs.google.com/spreadsheets/d/1PjaRp4bueQ3Cv14A7I696m0vjGCIX_kN9tIGh-V-Bkc/edit).

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Context

@o-fernandez requested this be added for the release! 
Screen Shot 2021-05-19 at 7 22 43 PM

## Summary

add check for video posters to prepublish checklist. when there is no poster present on a video check is present and will take user to poster media button in video accessibility panel on click.

## Relevant Technical Choices

Uses existing highlight structure, just adding a new ref for the video poster button

## User-facing changes

Adds a new check to the PPC that is triggered whenever video has no poster source. 

## Testing Instructions

Upload a video without a poster image (honestly I couldn't get this to trigger, I followed [these steps]( https://github.com/google/web-stories-wp/issues/7571):
> Click on "Upload" to open WP media modal
> Upload video
> After upload, insert the video to the story
> See how video poster is successfully being generated and attached on the video
> See how the video in the media library on the left is not being updated to include the poster (unless you refresh the page) 

but when you do get a video in the editor without a poster it'll show you a new **high priority** check

poster ppc

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7613

### CODE REVIEW

File: assets/src/edit-story/app/prepublish/constants.js [hunk 1]
```diff
@@ -636,5 +636,13 @@ export const MESSAGES = {
         </ul>
       ),
     },
+    VIDEO_MISSING_POSTER: {
```
Comment (swissspidy) [inline assets/src/edit-story/app/prepublish/constants.js:626]: As per https://github.com/google/web-stories-wp/issues/7571#issuecomment-843486280 there is already `MISSING_VIDEO_POSTER` some lines above...

Comment (BrittanyIRL) [reply assets/src/edit-story/app/prepublish/constants.js:626]: that's for the whole story, this is specific to videos in the story

Comment (swissspidy) [reply assets/src/edit-story/app/prepublish/constants.js:626]: Do you mean `MISSING_POSTER`? What I am referring to is `MISSING_VIDEO_POSTER`.

https://github.com/google/web-stories-wp/blob/c646e5bd76d3825d7ef355d1720f4d4ad741b6d9/assets/src/edit-story/app/prepublish/constants.js#L194-L207

That's not currently used anywhere actually. As per https://github.com/google/web-stories-wp/issues/7571#issuecomment-843486280 it seems like it was never properly implemented.

File: assets/src/edit-story/app/prepublish/constants.js [hunk 2]
```diff
@@ -190,19 +190,6 @@ export const MESSAGES = {
         </ul>
       ),
     },
-    MISSING_VIDEO_POSTER: {
```
Comment (BrittanyIRL) [inline assets/src/edit-story/app/prepublish/constants.js:193]: duplicative

File: assets/src/edit-story/app/prepublish/constants.js (hunk 1 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/app/prepublish/constants.js:626]: fixed, removed duplicate. kept the part i added since it is part of the media group

**Review by maxyinger — APPROVED** [review]
lgtm

**Review by embarks — APPROVED** [review]

**Review by samwhale — APPROVED** [review]
Looking at the tests I am confident that this will work in the event that the `featured image` of the video doesn't exist that we will see this error. however - i think wordpress is adding it in automatically in many cases

File: assets/src/edit-story/app/prepublish/constants.js [hunk 3]
```diff
@@ -636,5 +623,18 @@ export const MESSAGES = {
         </ul>
       ),
     },
+    VIDEO_MISSING_POSTER: {
+      MAIN_TEXT: __('Add poster image to every video', 'web-stories'),
```
Comment (BrittanyIRL) [inline assets/src/edit-story/app/prepublish/constants.js:627]: moving copy to media section of constants

**Comment by BrittanyIRL** [comment]
Pascal's suggestion on how to trigger this manually works, if anyone wants to test it: 
open the editor devtools (cmd+option+shift+j) and manually remove the poster from video resource, load that data.

**Comment by csossi** [comment]
Verified in QA env

File: assets/src/edit-story/components/inspector/karma/prepublishSelect.karma.js [hunk 4]
```diff
@@ -384,4 +384,83 @@ describe('Pre-publish checklist select offending elements onClick', () => {
       await fixture.snapshot('document tab opened by checklist panel');
     });
   });
+
+  it('should open the design panel video processing section and focus the reprocess video button when using mouse', async () => {
```
Comment (swissspidy) [inline assets/src/edit-story/components/inspector/karma/prepublishSelect.karma.js:388]: @BrittanyIRL these tests seem to have the wrong names?

Comment (BrittanyIRL) [reply assets/src/edit-story/components/inspector/karma/prepublishSelect.karma.js:388]: oh yup, hey at least the tests are _right_ outside of the names. I'll update this text in my next PR.

Comment (BrittanyIRL) [reply assets/src/edit-story/components/inspector/karma/prepublishSelect.karma.js:388]: updated in with this other test clean up: https://github.com/google/web-stories-wp/pull/7497/commits/60d1d8009b044912ccf85d4a225cad03f2a3456c

**Pull request merged**